jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdAuthenticator.java

```diff
@@ -494,7 +479,80 @@ protected String getChallengeUri(HttpServletRequest request)
     @Override
     public boolean secureResponse(ServletRequest req, ServletResponse res, boolean mandatory, User validatedUser)
     {
-        return true;
+        return req.isSecure();
+    }
+
+    private UriRedirectInfo removeAndClearCsrfMap(HttpSession session, String csrf)
+    {
+        @SuppressWarnings("unchecked")
+        Map<String, UriRedirectInfo> csrfMap = (Map<String, UriRedirectInfo>)session.getAttribute(CSRF_MAP);
+        if (csrfMap == null)
+            return null;
+
+        UriRedirectInfo uriRedirectInfo = csrfMap.remove(csrf);
+        csrfMap.clear();
+        return uriRedirectInfo;
+    }
+
+    private Map<String, UriRedirectInfo> ensureCsrfMap(HttpSession session)
+    {
+        @SuppressWarnings("unchecked")
+        Map<String, UriRedirectInfo> csrfMap = (Map<String, UriRedirectInfo>)session.getAttribute(CSRF_MAP);
+        if (csrfMap == null)
+        {
+            // Create a custom Map so we can only have a limited number of request URIs saved.
+            csrfMap = new LinkedHashMap<>()
+            {
+                private static final int MAX_SIZE = 64;
+
+                @Override
+                protected boolean removeEldestEntry(Map.Entry<String, UriRedirectInfo> eldest)
```

Comment:
```suggestion
                protected boolean removeOldestEntry(Map.Entry<String, UriRedirectInfo> eldest)
```

Reply:
@janbartel This overrides an existing method on `LinkedHashMap`. 
So I don't think I can change the name of this to `removeOldestEntry`.

Reply:
Hah! LinkedHashMap is wrong: "eldest" is just for people, "oldest" is for things.